java/org/contikios/cooja/Simulation.java

```diff
@@ -150,68 +155,133 @@ public Simulation(Cooja cooja, long seed) {
     this.cooja = cooja;
     randomGenerator = new SafeRandom(this);
     randomSeed = seed;
-  }
-
-  @Override
-  public void run() {
-    assert isRunning : "Did not set isRunning before starting";
-    lastStartRealTime = System.currentTimeMillis();
-    lastStartSimulationTime = getSimulationTimeMillis();
-    speedLimitLastRealtime = lastStartRealTime;
-    speedLimitLastSimtime = lastStartSimulationTime;
+    this.simulationThread = new Thread(() -> {
+      boolean isSimulationRunning = false;
+      boolean isAlive = true;
+      do {
+        EventQueue.Pair nextEvent = null;
+        try {
```

Comment:
Push the try into the place where it is needed.

Reply:
Inside the for-loop instead of outside? I can change this but we do want the loop to exit if there is an exception. Or do you mean separate it into two try - one for the blocking queue and one for running the simulation events?